alacritty_terminal/src/input.rs

```diff
@@ -692,6 +696,17 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
             self.mouse_report(code, ElementState::Released, modifiers);
             return;
         } else if let (Some(point), true) = (point, button == MouseButton::Left) {
+            let mouse_state = self.mouse_state(point);
+            self.update_mouse_cursor(mouse_state);
+            match mouse_state {
+                MouseState::Url(url) => {
+                    let url_bounds = url.linear_bounds(self.ctx.terminal());
+                    self.ctx.terminal_mut().set_url_highlight(url_bounds);
+                },
+                _ => {
+                    self.ctx.terminal_mut().reset_url_highlight();
+                }
```

Comment:
Why would we need this? This code is just to make sure that when the selection is cleared by clicking, that URLs are shown again, right? We're never concerned about removing URLs when the selection is cleared (since it shouldn't exist in the first place).

So this can just be transformed into an `if let`.